File: xr18/app/src/main/java/com/mixer/xr18/app/AppExecutors.java
package com.mixer.xr18.app;

import java.util.concurrent.ExecutorService;
import java.util.concurrent.Executors;

/**
 * Shared executor service for the app layer.
 */
public class AppExecutors {
    private static final ExecutorService executor = Executors.newSingleThreadExecutor();

    public static ExecutorService getExecutor() {
        return executor;
    }

    public static void shutdown() {
        executor.shutdownNow();
    }
}

File: xr18/app/src/main/java/com/mixer/xr18/app/AppExecutors.kt
package com.mixer.xr18.app

import kotlinx.coroutines.CoroutineScope
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.SupervisorJob

object AppExecutors {
    private val supervisorJob = SupervisorJob()
    val scope: CoroutineScope = CoroutineScope(Dispatchers.Main + supervisorJob)
}